src/state/url-sync-mixin.ts

```diff
@@ -62,16 +68,22 @@ export const urlSyncMixin = <
             if (DEBUG) {
               console.log("remove state", ev.detail.dialog);
             }
-            this._ignoreNextPopState = true;
-            mainWindow.history.back();
+            if (history.length) {
+              this._ignoreNextPopState = true;
+              historyPromise.promise = new Promise((resolve) => {
```

Comment:
I'm a big fan of adding cleanup where we also set the promise. So specifyin a 'then' here to cleanup historyPromise and historyResolve.